src/extension/extension.ts

```diff
@@ -451,8 +451,11 @@ export class RunmeExtension {
     }
 
     if (kernel.isFeatureOn(FeatureName.RequireStatefulAuth)) {
-      const forceLogin = kernel.isFeatureOn(FeatureName.ForceLogin)
-      context.subscriptions.push(new StatefulAuthProvider(context, uriHandler))
+      const statefulAuthProvider = new StatefulAuthProvider(context, uriHandler)
+      context.subscriptions.push(statefulAuthProvider)
+
+      const secretsAuthed = await statefulAuthProvider.bootstrapFromToken()
```

Comment:
Should this run if one's already signed in? More so, should a token be processed if there's a valid session?

The problem is that if a token was previously used to successfully bootstrap but skipping token deletion is now invalid?

Reply:
Solved. Now, I'm checking the sessions first.